controller/hydrator/hydrator.go

```diff
@@ -272,6 +290,25 @@ func (h *Hydrator) hydrate(logCtx *log.Entry, apps []*appv1.Application) (string
 	repoURL := apps[0].Spec.SourceHydrator.DrySource.RepoURL
 	syncBranch := apps[0].Spec.SourceHydrator.SyncSource.TargetBranch
 	targetBranch := apps[0].Spec.GetHydrateToSource().TargetRevision
+
+	// Disallow hydrating to the repository root.
+	// Hydrating to root would overwrite or delete files at the top level of the repo,
+	// which can break other applications or shared configuration.
+	// Every hydrated app must write into a subdirectory instead.
+	//
+	// IMPORTANT: this is a second layer of validation because
+	// the hydration destination might differ
+	// (e.g., HydrateTo.TargetBranch pointing at root, or a bad config slipping past).
```

Comment:
```suggestion
```

I think we can remove this part of the comment.

There's no `path` field in `hydrateTo`, the path always comes from `syncSource`.

The kubebuilder validation provides one line of defense, but it's very easy to bypass with a value like `dir/..`.